### PR TITLE
FIX-2296 Move styled buttons

### LINKED ISSUE #2296: Add one styled component for buttons
There are multiple button styles in the application that should be replaced with a single styled component. This component should be stored in the styled component directory of the project.

### PR DESCRIPTION
**Pull request by eliasbruvik**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #2296

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

Moved the common styled buttons to StyledComponents.

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [ ] API
* [ ] WITSML
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx [hunk 1]
```diff
@@ -60,14 +60,14 @@ const EditNumber = (props: EditNumberProps): ReactElement => {
           ) : null
         }
       />
-      <StyledButton
-        variant={"ghost"}
+      <OutlinedIconButton
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx]: This should be changed to `StyledGhostButton` to have consistent styling for all icon buttons.

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLogCurveInfo.tsx [hunk 2]
```diff
@@ -213,8 +213,8 @@ const EditSelectedLogCurveInfo = (
               dropdownHeight={600}
             />
           </StartEndIndex>
-          <StyledButton
-            variant={"ghost"}
+          <OutlinedIconButton
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLogCurveInfo.tsx]: This should be changed to StyledGhostButton to have consistent styling for all icon buttons.

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLogCurveInfo.tsx [hunk 3]
```diff
@@ -226,7 +226,7 @@ const EditSelectedLogCurveInfo = (
             }
           >
             <Icon size={16} name={isEdited ? "arrowForward" : "sync"} />
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditSelectedLogCurveInfo.tsx]: Remove size, as it conflicts with the styling of other icon buttons.

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx [hunk 4]
```diff
@@ -60,14 +60,14 @@ const EditNumber = (props: EditNumberProps): ReactElement => {
...
+        variant={"ghost_icon"}
         color={"primary"}
         onClick={submitEditNumber}
         disabled={!isEdited}
       >
         <Icon size={16} name={"arrowForward"} />
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx]: Remove size, as it conflicts with the styling of other icon buttons.

File: Src/WitsmlExplorer.Frontend/components/StyledComponents/Buttons.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import styled from "styled-components";
+import { Colors, colors } from "styles/Colors";
+
+// A styled button to get the correct color when using the ghost or outlined variants (no fill)
+export const StyledGhostButton = styled(Button)<{ colors: Colors }>`
+  white-space: nowrap;
+  color: ${(props) => props.colors.infographic.primaryMossGreen};
+`;
+
+export const OutlinedIconButton = styled(Button)`
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/StyledComponents/Buttons.tsx]: I think we can remove this one, as having several types of icon buttons in the application is not necessary.

File: Src/WitsmlExplorer.Frontend/components/StyledComponents/Buttons.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import { Button } from "@equinor/eds-core-react";
+import styled from "styled-components";
+import { Colors, colors } from "styles/Colors";
+
+// A styled button to get the correct color when using the ghost or outlined variants (no fill)
+export const StyledGhostButton = styled(Button)<{ colors: Colors }>`
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/StyledComponents/Buttons.tsx]: Is IconButton a more descriptive name?

**Review by janmarius — CHANGES_REQUESTED** [review]
Since this PR does not create one generic Button for the whole application, I think we should create another issue to solve this. This will make it a lot easier to use different variations of the EDS-Button in different places in the application.

I made a few comments, but overall, it looks good. Nice cleanup.

**Comment by eliasbruvik** [comment]
Jan-Marius wants to handle everything in a Buttons-component, which requires more changes and makes this PR redundant.

**Pull request closed without merging**